python/cudf/cudf/core/dataframe.py

```diff
@@ -365,12 +365,13 @@ def _setitem_tuple_arg(self, key, value):
         try:
             columns_df = self._frame._get_columns_by_label(key[1])
         except KeyError:
-            if not self._frame.empty and isinstance(key[0], slice):
+            was_empty = self._frame.empty
```

Comment:
Are you caching this for performance, or just to shorten how much you're writing? If it's for the former, should we change this to a cached property?

Reply:
Since this code path modifies `self._frame._data` inplace via `insert`, I needed to save this because the inserted column label may no longer be the same type as `._data.label_dtype` if `._data` was first empty.

i.e. the below 

```python
            if was_empty:
                # TODO: Create a cudf_to_pandas_dtype function to use here
                self._frame._data.label_dtype = None
```

But I realized this logic better lives in `ColumnAccessor.insert`